package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +746,451 @@ def __setstate__(self, state):
+                          'velocity': kwargs.get('velocityunit', None),
+                          'force': kwargs.get('forceunit', None)}
+
+        # Pull out various keywords to store metadata in 'h5md' group
+        self.author = kwargs.pop('author', 'N/A')
+        self.author_email = kwargs.pop('author_email', None)
+        self.creator = kwargs.pop('creator', 'MDAnalysis')
+        self.creator_version = kwargs.pop('creator_version', mda.__version__)
+
+    def _write_next_frame(self, ag):
+        """Write information associated with ``ag`` at current frame
+        into trajectory
+
+        Parameters
+        ----------
+        ag : AtomGroup or Universe
+
+        """
+        try:
+            # Atomgroup?
+            ts = ag.ts
+        except AttributeError:
+            try:
+                # Universe?
+                ts = ag.trajectory.ts
+            except AttributeError:
+                errmsg = "Input obj is neither an AtomGroup or Universe"
+                raise TypeError(errmsg) from None
+
+        if ts.n_atoms != self.n_atoms:
+            raise IOError("H5MDWriter: Timestep does not have"
+                          " the correct number of atoms")
+
+        # Opens H5MD file, initializes 'h5md' group, and checks
+        # if unitcell is periodic.
+        # This should only be called once when first timestep is read.
+        if self.h5md_file is None:
+            self._determine_units(ag)
+            self._open_file()
+            self._initialize_hdf5_datasets(ts)
+
+        # skips _write_next_timestep() so you can write an empty Universe
+        #if 'edges' not in self.traj['box'] and not any([group in self.traj for group in ('position', 'velocity', 'force')]):
+            #return
+        return self._write_next_timestep(ts)
+
+    def _determine_units(self, ag):
+        """sets native units to the units of the parent file unless
+        'unit' keyword arguments are provided for custom units"""
+
+        if not self.convert_units:
+            return
+
+        # use units of parent file as native units
+        try:
+            # ag == AtomGroup
+            self.units = ag.universe.trajectory.units
+        except AttributeError:
+            # ag == Universe
+            self.units = ag.trajectory.units
+        # set user-inputted new native units
+        for key, value in self.new_units.items():
+            if value is not None:
+                if value not in self._unit_translation_dict[key]:
+                    raise RuntimeError(f"{value} is not a unit recognizable by"
+                                        " MDAnalyis. Allowed units are:"
+                                       f" {self._unit_translation_dict.keys()}"
+                                        " For more information on units, see"
+                                        " `MDAnalyis units`_.")
+                else:
+                    self.units[key] = self.new_units[key]
```

Comment:
I'm pretty sure `self.units` has to be in MDA units since all of the `convert_*_to_native()` methods pull from `self.units.values()` to get conversion factors